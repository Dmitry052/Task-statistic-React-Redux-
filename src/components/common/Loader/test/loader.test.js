import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { Loader } from './../index';

configure({ adapter: new Adapter() });

describe('<Loader />', () => {
  const tree = shallow(
    <Loader styleBlock={'some-test-class'} />
  );

  it('should contain 1 div elements', () => {
    expect(tree.find('div').length).toBe(1);
  });

  it('should contain 1 i elements', () => {
    expect(tree.find('div').children().length).toBe(1);
  });

  it('Matches the snapshot HeaderTable component', () => {
    expect(toJson(tree)).toMatchSnapshot();
    tree.unmount();
  });
});
