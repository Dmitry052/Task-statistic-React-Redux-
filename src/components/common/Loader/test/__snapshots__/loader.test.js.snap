// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Matches the snapshot HeaderTable component 1`] = `
<div
  className="some-test-class"
>
  <i
    aria-hidden="true"
    className="fa fa-spinner fa-pulse fa-4x fa-fw"
  />
</div>
`;
